openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/OASModifier.java

```diff
@@ -286,7 +340,7 @@ private static String updateParameters(Map<String, Schema> modifiedSchemas, Stri
                 continue;
             }
             // check given parameter has name which similar to component schema name
-            if (modifiedSchemas.containsKey(modifiedPathParam)) {
+            if (modifiedSchemasName.contains(modifiedPathParam)) {
```

Comment:
We don't need to handle the conflicts here. Refer to this [comment](https://github.com/ballerina-platform/ballerina-library/issues/6897#issuecomment-2304033370)

Reply:
Address parameterName handing via https://github.com/ballerina-platform/openapi-tools/pull/1762/commits/5cf7b9fbe11ec7e1eff739aa1b3b6068a1eba081